recipes/default.rb

```diff
@@ -19,13 +18,14 @@
 # using chef-search, which do have the same networks, and create hosts files on this node for those with their public key
 # so they can actually connect to each other
 node['tincvpn']['networks'].each do |network_name, network|
-  raise "You need to set the host name for the tinc network #{network_name} in ['tincvpn']['networks'][#{network_name}]['network']['host']['name']" if network['host']['name'].nil?
-  raise 'You defined switch as you mode, but also defined subnets - this is now allowed by tinc' if !node['tincvpn']['networks'][network_name]['host']['subnets'].empty? && network['network']['mode'] == 'switch'
+  if !node['tincvpn']['networks'][network_name]['host']['subnets'].empty? && network['network']['mode'] == 'switch'
+    raise 'You defined switch as you mode, but also defined subnets - this is now allowed by tinc'
+  end
 
   directory "/etc/tinc/#{network_name}"
   directory "/etc/tinc/#{network_name}/hosts"
-  local_host_name = node['tincvpn']['networks'][network_name]['host']['name']
-  local_host_path = "/etc/tinc/#{network_name}/hosts/#{local_host_name.gsub('-', '_')}"
+  local_host_name = node['hostname'].gsub('-', '_')
```

Comment:
AFAICS this would break my setup, since the hostname is set using the node['tincvpn']['networks'][network_name]['host']['name'] variable.

We have to ensure that we do not break existing setups, so i suggest if `node['tincvpn']['networks'][network_name]['host']['name']` is set, use it - if not, use `node['hostname']`

Reply:
> ... since the hostname is set using the node['tincvpn']['networks'][network_name]['host']['name'] variable.

I'm not sure to understand as this attribute is no more used in this cookbook, that was the behaviour before this PR.

This `node['hostname']` is populated by Ohai automatically.

> We have to ensure that we do not break existing setups, so i suggest if node['tincvpn']['networks'][network_name]['host']['name'] is set, use it - if not, use node['hostname']

Ok, let's do it like this, I understand your concern.

Reply:
In addition of using `node['tincvpn']['networks'][network_name]['host']['name']` or `node['hostname']`, what about to add a warning when comparing the 2 values?

Something like that :

```ruby
unless node['tincvpn']['networks'][network_name]['host']['name'] == node['hostname']
  Chef::Log.warn("Host name from tincvpn attributes (#{node['tincvpn']['networks'][network_name]['host']['name'].inspect}) differs with node's hostname (#{node['hostname'].inspect}).")
end
```
_(Or we can set it as debug if you prefer)_

With that in place, when you'll use it, if the warning never appears, it will mean that you no more need to set the host name attribute.

Reply:
Comparing and warning is an nice idea, but we should still enforce the old value as long as it is present

Reply:
Yep, and that's what I did at line 32.